### PR TITLE
[JIT] register list prim ops cleanup

### PR DESCRIPTION
**Pull request by eellison**
This is a follow up from https://github.com/pytorch/pytorch/pull/34520, which removed specialized list ops. This removes templating from list ops. 

it also has one minor other change, which is to move `aten::len(t[]) -> int` to `aten::len(Any[]) -> int` so that heterogenous tuples can be called with `len()`.

### CODE REVIEW

File: torch/csrc/jit/runtime/register_prim_ops.cpp [hunk 1]
```diff
@@ -1191,20 +1191,18 @@ void setItem(const c10::List<T>& list, int64_t idx, T&& value) {
   list.set(normalized_idx, std::move(value));
 }
 
-template <typename T>
 int listAppend(Stack& stack) {
-  T el = pop(stack).to<T>();
-  c10::List<T> list = pop(stack).to<c10::List<T>>();
+  IValue el = pop(stack).to<IValue>();
+  c10::List<IValue> list = pop(stack).to<c10::List<IValue>>();
```
Comment (driazati) [inline torch/csrc/jit/runtime/register_prim_ops.cpp]: I don't know how `c10::List` works under the hood, but if you have a `c10::List<int64_t>` and try to use it as `c10::List<IValue>` here won't you get some covariant type issues?

File: torch/csrc/jit/runtime/register_prim_ops.cpp [hunk 2]
```diff
@@ -2008,105 +1989,83 @@ RegisterOperators reg2({
...
+    Operator(
+        "aten::mul_.t(t[](a!) l, int n) -> t[](a!)",
+        listMulIntLeftInPlace,
+        aliasAnalysisFromSchema()),
 
-    CREATE_GENERIC_LIST_OPS("t", IValue),
+    // registered as Any[] so that heterogenous tuples can be called with len()
+    Operator(
+        "aten::len.t(Any[] a) -> int",
+        listLen<IValue>,
```
Comment (driazati) [inline torch/csrc/jit/runtime/register_prim_ops.cpp]: The template parameter on `listLen` can also be deleted

Comment (eellison) [reply torch/csrc/jit/runtime/register_prim_ops.cpp]: that would be a BC incompat change for mobile

File: torch/csrc/jit/runtime/register_prim_ops.cpp (hunk 1 above)
Comment (eellison) [reply torch/csrc/jit/runtime/register_prim_ops.cpp]: This is exactly how the code was working previously. There's no functional difference.

File: torch/csrc/jit/runtime/register_prim_ops.cpp (hunk 2 above)
Comment (driazati) [reply torch/csrc/jit/runtime/register_prim_ops.cpp]: Why is that? it looks like `listLen` isn't used outside of this file

Comment (eellison) [reply torch/csrc/jit/runtime/register_prim_ops.cpp]: oop yea you're right.

**Review by driazati — APPROVED** [review]

**Pull request closed without merging**